### PR TITLE
Final Page Builder TOC and removal of 

### PR DESCRIPTION
**Pull request by bdenham**
## Purpose of this pull request

This pull request (PR) finishes the last directory and file name changes made to the Page Builder repo. It also removes the entry for `magento/magento2-page-builder-data-migration` from the Docfile.yml. I moved the handful of files we were pulling from the page builder migration repo, into the main page builder docs repo. This should help simplify our pipeline a little. 

## Affected DevDocs pages

Several Page Builder topics.

I've covered the Page Builder docs reorg in a previous PR.

### CODE REVIEW

File: src/_data/toc/rest-api.yml [hunk 1]
```diff
@@ -40,15 +40,15 @@ pages:
           children:
 
             - label: Admin REST endpoints
-              url: https://magento.redoc.ly/2.4.2-admin/
+              url: https://magento.redoc.ly/2.4.1-admin/
```
Comment (dshevtsov) [inline src/_data/toc/rest-api.yml:43]: This looks like an unintended change

Comment (bdenham) [reply src/_data/toc/rest-api.yml:43]: Oops. Yeah that snuck in...Got it.

File: src/_data/toc/rest-api.yml [hunk 2]
```diff
@@ -44,11 +44,11 @@ pages:
               include_versions: ["2.4"]
 
             - label: Customer REST endpoints
-              url: https://magento.redoc.ly/2.4.2-customer/
+              url: https://magento.redoc.ly/2.4.1-customer/
```
Comment (dshevtsov) [inline src/_data/toc/rest-api.yml:47]:
```suggestion
              url: https://magento.redoc.ly/2.4.2-customer/
```

File: src/_data/toc/rest-api.yml [hunk 3]
```diff
@@ -44,11 +44,11 @@ pages:
...
               include_versions: ["2.4"]
 
             - label: Guest REST endpoints
-              url: https://magento.redoc.ly/2.4.2-guest/
+              url: https://magento.redoc.ly/2.4.1-guest/
```
Comment (dshevtsov) [inline src/_data/toc/rest-api.yml:51]:
```suggestion
              url: https://magento.redoc.ly/2.4.2-guest/
```

**Review by dshevtsov — CHANGES_REQUESTED** [review]

**Review by dshevtsov — APPROVED** [review]

**Comment by bdenham** [comment]
running tests

**Comment by Leland** [comment]
Hi @bdenham, a heads up that this change didn't add any kind of redirects, so folks looking for this on search engines are currently getting served 404s. Try Googling for "How to upgrade content types" for, instance:  https://www.google.com/search?q=%22How+to+upgrade+content+types%22

I speak as one such individual who had to really dig to find what he was looking for :)

**Pull request merged**